lib/ecto/adapters/postgres/connection.ex

```diff
@@ -682,6 +682,13 @@ if Code.ensure_loaded?(Postgrex) do
       "ALTER TABLE #{quote_table(constraint.prefix, constraint.table)} DROP CONSTRAINT #{quote_name(constraint.name)}"
     end
 
+    def execute_ddl({:create, %View{} = view}) do
+      {query, params, _key} = Ecto.Query.Planner.prepare(view.query, :all, Ecto.Adapters.Postgres, 0)
```

Comment:
Makes sense. To be clear, we should keep the code as it is today, we just need to move it to outside of the adapters. The adapters should receive the already normalized query. The normalization should happen either in Ecto.Migration or Ecto.Migration.Runner. :)